server/controllers/submission.js

```diff
@@ -0,0 +1,146 @@
+    if (contest) {
+        const createdSubmission = await Submission.create({
+            creator,
+            contest_id,
+            files,
+            is_active,
+            date_Created,
+        });
+        //Creating a finding and updating contest to apply submission id to array within contest
+        if (createdSubmission) {
+            let submissionArr = contest.submissions;
+            submissionArr.push(createdSubmission._id);
+            contest.submissions = submissionArr;
+
+            const title =contest.title;
+            const price =contest.price;
+            const description = contest.description;
+            const end_date = contest.end_date
+            const images = contest.images;
+            const submissions = contest.submissions;
+            //
+            let createdContest = await Contest.findByIdAndUpdate(
+                contest_id,
+                {
+                    title,
+                    price,
+                    description,
+                    end_date,
+                    images,
+                    submissions,
+                },
+                { new: true }
+            );
+            if (createdContest) {
+                console.log("successfully updated contest");
+            } else {
+                console.log("failed to update contest");
+            }
+
+            res.status(201).json({
+                success: {
+                    submission: createdSubmission,
+                },
+            });
+        } else {
+            res.status(500);
+            throw new Error(
+                "Submission could not be processed at this time, Please Try Again"
+            );
+        }
+    } else {
+        res.status(500);
+        throw new Error(
+            "Contest not found to allow submission"
+        )
+    }
+});
+
+exports.getAllSubmissions = asyncHandler(async (req, res, next) => {
+   //Grabs necessary data from req params
+    console.log(req.params);
+    const contest_id = req.params.id;
+    const Creator = req.params.creator;
+    console.log("contest id",contest_id);
+    console.log("creator", Creator);
+
+    /**
+     * Test to check if ID exists
+     */
+    if(!contest_id){
+        res.status(404);
+        throw new Error("No contest id given");
+    }else if(!Creator){
+        res.status(404);
+        throw new Error("No creator id given");
+    }
+    /**
+     * Does a check on the specific contest to see if the user is the contest owner,
+     * if the user is the contest owner. output is all submissions within that contest
+     * if the user is not the contest owner shows all submissions related to the user.
+     */
+    else{
+        let contest = await Contest.findById(contest_id);
+        
+        if (!contest) {
+            res.status(404);
+            throw new Error("No contest found for given id");
+        }
+
+        if(contest.creator.toString() === Creator.toString()){
```

Comment:
are we comparing User objects here? i would suggest comparing IDs instead

Reply:
That point is comparing IDs

Reply:
oh i see so `.creator` attribute is an ID? okay not a huge issue but best practice is to rename this to `creatorId` cause it seems like you're comparing objects

Reply:
The .creator attribute was named within the model by someone else! Sorry for the confusion !